test/integration/servicebus/queue.test.js

```diff
@@ -18,9 +18,10 @@ describe('ServiceBus', function() {
       test.client = new AMQPClient(Policy.ServiceBusQueue);
     });
 
-    afterEach(function() {
+    afterEach(function(done) {
       return test.client.disconnect().then(function() {
         test.client = undefined;
+        done();
```

Comment:
I was encountering another issue (the fix below to store a local copy of connection and attach the disconnect handler _after_ calling removeAllListeners fixed that), so technically I don't need this anymore. _But_ isn't it the correct way to do things since I'm calling an async method in here?